crowbar_framework/app/models/service_object.rb

```diff
@@ -1576,4 +1616,26 @@ def profile(name, &block)
       block.call
     end
   end
+
+  def equal_or_match(value, maybe_regexp)
+    operator = lambda do
+      op, check = maybe_regexp.split(" ", 2)
+      value.to_f.send(op.to_sym, check.to_f)
```

Comment:
I discussed this with @tboerger, but looks like that this is the expected behaviour for chef: '12.04.1'.to_f == 12.04, so '12.04' match with '== 12.04.1', but is not '< 12.04.1'.

But because we remove the '==' operator, (Chef do not provide it), if :platform_version is '12.04.1', it will not match with '12.04' restriction, and this is wrong IMHO.

To resolve that I will add a new check before the string comparison.

But I tend to disagree with the to_f approach for something that is not a number.